opentelemetry/metrics.go

```diff
@@ -0,0 +1,83 @@
+package opentelemetry
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregator/histogram"
+	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	processor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
+	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	selector "go.opentelemetry.io/otel/sdk/metric/selector/simple"
+)
+
+type ChainbridgeMetrics struct {
+	DepositEventCount metric.Int64Counter
+}
+
+func newChainbridgeMetrics(meter metric.Meter) *ChainbridgeMetrics {
+	return &ChainbridgeMetrics{
+		DepositEventCount: metric.Must(meter).NewInt64Counter(
+			"chainbridge.DepositEventCount",
+			metric.WithDescription("Number of deposit events across all chains"),
+		),
+	}
+}
+
+func initPrometheusMetrics(port uint64, path string) (*ChainbridgeMetrics, error) {
+	config := prometheus.Config{}
+	c := controller.New(
+		processor.NewFactory(
+			selector.NewWithHistogramDistribution(
+				histogram.WithExplicitBoundaries(config.DefaultHistogramBoundaries),
+			),
+			export.CumulativeExportKindSelector(),
+			processor.WithMemory(true),
+		),
+	)
+	exp, err := prometheus.New(config, c)
+	if err != nil {
+		return nil, err
+	}
+
+	global.SetMeterProvider(exp.MeterProvider())
+	go func() {
+		http.HandleFunc(path, exp.ServeHTTP)
+		_ = http.ListenAndServe(":"+strconv.Itoa(int(port)), nil)
+	}()
+
+	meter := c.Meter("")
```

Comment:
I'm not sure what this represents (I guess meter name), but it seems strange that it is an empty string.

Reply:
It basically uses the default name if empty string is provided.
I couldn't find if it made difference anywhere when it is changed.
I guess I could name it just "chainbridge" to confuse people.